src/bin/server.rs

```diff
@@ -1,4 +1,31 @@
+extern crate bincode;
+extern crate rustc_serialize;
+extern crate rpsrtsrs;
+use std::net::TcpListener;
+use std::io::Write;
+use std::ops::Deref;
+
+use rpsrtsrs::colors;
+use rpsrtsrs::state::{Game, Player};
+
+use bincode::SizeLimit;
+use bincode::rustc_serialize::encode;
 
 fn main() {
-    println!("This will be the server in the future!");
+    let game = Game{
+        players: vec![
+            Player{
+                color: colors::ORANGE,
+                units: vec![],
+            }]
+    };
+    let encoded: Vec<u8> = encode(&game, SizeLimit::Infinite).unwrap();
+
+    let socket_addr = "127.0.0.1:8080".to_string();
+    let tcp_listener = TcpListener::bind(socket_addr.deref()).unwrap();
```

Comment:
No because bind needs a type implementing [`ToSocketAddrs`](http://doc.rust-lang.org/nightly/std/net/trait.ToSocketAddrs.html) which is implemented for `&str`, but auto-coercion fails in this case. One could use `&*` see here https://www.reddit.com/r/rust/comments/3cce2h/tcplistenerbind_fails_when_passing_a_string/ for more information.